robyn/__init__.py

```diff
@@ -191,12 +191,13 @@ def startup_handler(self, handler: Callable) -> None:
     def shutdown_handler(self, handler: Callable) -> None:
         self._add_event_handler(Events.SHUTDOWN, handler)
 
-    def is_port_in_use(self, port: int) -> bool:
+    def get_socket(self, url: str, port: int) -> (bool, SocketHeld):
```

Comment:
Hey @VishnuSanal 👋 

This solution looks interesting. Could you explain what it is doing?

And also add a docstring to the function signature?

Reply:
Hey @VishnuSanal 

Could you please explain what the function is doing? Reply as a comment here.

Reply:
@sansyrox hi. so, this function returns a tuple of whether the socket generation was successful and the actual socket, if successful. in the try block, we are checking whether the socket is already occupied (as before -- which gives us a true/false). we are also trying to create the socket using `SocketHeld`. which, upon failure goes to the except: block where we are returning false for the status & socket none.

our result can have 2 possibilities:

- true, <valid socket>
- false, none